### DISCUSSION TITLE
How / Where does the record returned from the DB get mapped to a plain JS object?

### DISCUSSION THREAD

**Discussion by anthonyorona** [gajus/slonik#521] [Q&A]
I am using slonik to handle some of my more complex queries. I would like to learn more about where the DB row gets mapped to a plain JS object

**Comment by nathanchapman** [comment]
Slonik is built on top of [pg](https://github.com/brianc/node-postgres/tree/master/packages/pg). Additional details here: https://github.com/gajus/slonik#pg-vs-slonik

`pg` handles the conversion from a db result to a JS object.

Here's a good starting point in `slonik` to better understand how the query is executing and the additional metadata on the query result:
https://github.com/gajus/slonik/blob/1228dba2fafcb002a7a78b76864011c08a388453/src/connectionMethods/query.ts#L10-L41

If you want to modify the behavior of slonik, you can use an [interceptor](https://github.com/gajus/slonik#interceptors), specifically a [result parser interceptor](https://github.com/gajus/slonik#result-parser-interceptor) or some of the [community interceptors](https://github.com/gajus/slonik#community-interceptors) to modify how result objects are built
